### PR TITLE
Update CI

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -61,41 +26,40 @@ jobs:
...
+        run: cargo doc --document-private-items
+
+  fmt:
+    name: Rustfmt
+    timeout-minutes: 30
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Check formatting
+        run: cargo fmt --all -- --check
```
Comment (sellout) [inline .github/workflows/ci.yml:65]: Noticed that this loses `--color always`.

File: .github/workflows/lints-beta.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Beta lints
```
Comment (sellout) [inline .github/workflows/lints-beta.yml:1]: Switching from nightly to beta, and extracting it to run more selectively.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -61,41 +26,40 @@ jobs:
...
-          args: --verbose --target ${{ matrix.target }} --no-default-features
+        run: >
+          cargo build
+          --verbose
+          --target ${{ matrix.target }}
+          --no-default-features
 
-  doc-links:
-    name: Nightly lint
+  bitrot:
```
Comment (sellout) [inline .github/workflows/ci.yml:39]: New, to make sure we test all features _somewhere_.

**Review by sellout — APPROVED** [review]
utACK

Comments are just to make sure I understand intentional changes. So, ignore them if you concur.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (str4d) [reply .github/workflows/ci.yml:39]: Yep, and benches / examples.

**Pull request merged**